basic support for colors and bold/bright

PeterMitrano opened the pull request:
I ported this from #26 to rolling :)

# Motivation
I like to print things with color, using Colorama in python this is super easy.

```
from colorama import Fore, Style, init
import colorama
colorama.init(autoreset=True)
rospy.loginfo(Fore.RED + Style.BRIGHT + "bold and red!")
```

However, this would shows up in the rqt console with no color and lots of ugly ansi codes

# What this does

makes foreground/background color and bold work, via the standard model/view abstraction mechanisms. One pitfall could be the regex I came up with it. It's possible it has false positives.

Here's a screenshot of what it looks like for me with some test logs messages

![2020-09-08-002906_1909x203_scrot](https://user-images.githubusercontent.com/4010770/92433207-5bf79100-f16a-11ea-8abd-b1f7b3d2caa2.png)

Commit:
basic support for colors and bold/bright

To test, install `colorama` and run test/test_colorama_logging.py

Commit:
support more color codes

peter-mitrano-ar commented:
Another example , you can reproduce it by running this example from [ros2_control_demos](https://github.com/ros-controls/ros2_control_demos)

```bash
ros2 launch ros2_control_demo_example_7 r6bot_controller.launch.py gui:=false
```


<img width="1854" height="1048" alt="image" src="https://github.com/user-attachments/assets/0d796b10-ecc5-408a-a443-ad8395d6fae5" />

arne48 commented:
Hello @PeterMitrano ,
Thank you very much for contributing to this package.
Eye-candy is always a nice way to improve the user-experience.
So generally I would like to merge it.
I think it would also beneficial, if there is an option (tick-box) to control (enable/disable) this new behavior.
Would it be possible for you to extend your PR with this functionality?

ahcorde commented:
friendly ping @PeterMitrano

peter-mitrano-ar commented:
Yes, how about I add a checkbox on the top right.
Does anyone have a strong opinion about on by default or off by default? If it's off by default no one will ever discover it, but if it's on and someone finds it annoying or it doesn't work for them, they may not find/know to look for a new checkbox to turn if off. :shrug: 
Hopefully I will find the time soon :)

arne48 commented:
I my opinion on-by-default would be fine.
Based on the regex, I think people would have to be very "unlucky" to hit a false positive in which case the behavior can be turned off via the checkbox.
So looking forward to your PR :)

Commit:
add checkbox in the settings

Also:
 - redo the layout of the settings to make it simpler
 - add sleep to ensure the logs in the test script actually get sent

PeterMitrano commented:
We now have a checkbox in the settings menu :) 

Only a bit of feature creep.... in adding the checkbox to the settings I noticed that the layout was a bit silly. It was a vertical layout, then a separate horizontal layout for each setting. I think a Form layout is a much simpler choice that looks almost identical.

I also made it so you can select individual items instead of whole rows... maybe just my opinion but usually you don't care about selecting (copying) the severity, stamp, location, etc... you still can of course, but you have to select it manually.

It was slightly more complicated to add a new setting than I would have thought. Maybe some room for future refactoring improvement there too.

arne48 commented:
Thank you very much.
I also like the additional changes you mentioned/implemented.
At a first glance it looks good and I am sure you already thoroughly tested it.
Now I will also just give it a quick test and then we can merge/publish it.

ahcorde commented:
Pulls: ros-visualization/rqt_console#54
Gist: https://gist.githubusercontent.com/ahcorde/ba8717cbcf7b92c821360e6e0dec8aea/raw/ec775cf4687bbd480a6e98139fb5f09f48491ca1/ros2.repos
BUILD args:  --packages-above-and-dependencies rqt_console
TEST args:  --packages-above rqt_console
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/17732
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=26476)](http://ci.ros2.org/job/ci_linux/26476/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=20106)](http://ci.ros2.org/job/ci_linux-aarch64/20106/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=6239)](http://ci.ros2.org/job/ci_linux-rhel/6239/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=26182)](http://ci.ros2.org/job/ci_windows/26182/)

PeterMitrano commented:
Fat fingered deleting the branch... anyways yes the test relies on `colorama`, I just did that to make it simple. I can replace it with manual escape codes to remove the dependency.

Commit:
replace colorama dependency with manual ansi codes

PeterMitrano commented:
Should be good now :) 
By the way, this time I decided to test pixi for running it. Worked great, but I noticed the icons for "settings" and "clear" don't show up anymore ... if anyone knows why, I would love to know.

arne48 commented:
Thank you very much for your PR @PeterMitrano
I couldn't recreate the icons issue. If you could append an image and we can recreate it then we can fix it before publishing the new version.
Everything looks good so we can merge and publish your contribution when the icons issue is clearified.
